### PR TITLE
support persistent remappable keys feature

### LINKED ISSUE #832: Feature Persistent Remappable Action {1C00}
It would be great to have persistent key or button remappings, which survive the device being turned off/on if Solaar is not running. See also https://github.com/pwr-Solaar/Solaar/issues/519#issuecomment-652018424 and related to https://github.com/pwr-Solaar/Solaar/issues/582#issue-530642547

**Information**
- Output of `solaar show` for the target device (if applicable):

```
> M720 Triathlon Multi-Device Mouse
>      Codename     : M720 Triathlon
>      Kind         : mouse
>      Wireless PID : 405E
>      Protocol     : HID++ 4.5
>      Polling rate : 8 ms (125Hz)
>      Serial number: C00C3906
>         Bootloader: BOT 34.00.B0004
>           Firmware: MPM 04.02.B0009
>              Other: 
>      The power switch is located on the base.
>      Supports 35 HID++ 2.0 features:
>          0: ROOT                   {0000}   
>          1: FEATURE SET            {0001}   
>          2: DEVICE FW VERSION      {0003}   
>          3: DEVICE NAME            {0005}   
>          4: WIRELESS DEVICE STATUS {1D4B}   
>          5: RESET                  {0020}   
>          6: CRYPTO ID              {0021}   
>          7: DEVICE FRIENDLY NAME   {0007}   
>          8: BATTERY STATUS         {1000}   
>          9: CHANGE HOST            {1814}   
>         10: HOSTS INFO             {1815}   
>         11: REPROG CONTROLS V4     {1B04}   
>             Actions: {'80': 80, '81': 81, '82': 82, '83': 83, '86': 86, '91': 91, '93': 93, '208': 208}
>         **12: PERSISTENT REMAPPABLE ACTION {1C00}**   
>         13: POINTER SPEED          {2205}   
>             Pointer Speed: 1.0
>             Sensitivity (Pointer Speed): 256
>         14: VERTICAL SCROLLING     {2100}   
>             Roller type: 3G
>             Ratchet per turn: 24
>             Scroll lines: 0
>         15: DFUCONTROL UNSIGNED    {00C1}   
>         16: DFUCONTROL SIGNED      {00C2}   
>         17: DEVICE RESET           {1802}   internal, hidden
>         18: unknown:1803           {1803}   internal, hidden
>         19: CONFIG DEVICE PROPS    {1806}   internal, hidden
>         20: OOBSTATE               {1805}   internal, hidden
>         21: unknown:1813           {1813}   internal, hidden
>         22: unknown:1830           {1830}   internal, hidden
>         23: unknown:1861           {1861}   internal, hidden
>         24: unknown:1890           {1890}   internal, hidden
>         25: unknown:1891           {1891}   internal, hidden
>         26: unknown:18A1           {18A1}   internal, hidden
>         27: unknown:1DF3           {1DF3}   internal, hidden
>         28: unknown:1E00           {1E00}   hidden
>         29: unknown:1EB0           {1EB0}   internal, hidden
>         30: unknown:18B1           {18B1}   internal, hidden
>         31: unknown:1850           {1850}   internal, hidden
>         32: unknown:1F03           {1F03}   internal, hidden
>         33: unknown:18C0           {18C0}   internal, hidden
>         34: HIRES WHEEL            {2121}   
>             Multiplier: 8
>             Has invert
>               Normal wheel motion
>             Has ratchet switch
>               Free wheel mode
>             High resolution mode
>             HID notification
>             High Resolution Wheel Invert: False
>             Wheel Resolution: True
>      Has 9 reprogrammable keys:
>          0: Left Button               , default: Left Click                  => Left Button               
>              mse, reprogrammable, pos:0, group:1, gmask:1
>          1: Right Button              , default: Right Click                 => Right Button              
>              mse, reprogrammable, pos:0, group:1, gmask:1
>          2: Middle Button             , default: Mouse Middle Button         => Middle Button             
>              mse, reprogrammable, divertable, persistently divertable, pos:0, group:2, gmask:3
>          3: Back Button               , default: Mouse Back Button           => Back Button               
>              mse, reprogrammable, divertable, persistently divertable, pos:0, group:2, gmask:3
>          4: Forward Button            , default: Mouse Forward Button        => Forward Button            
>              mse, reprogrammable, divertable, persistently divertable, pos:0, group:2, gmask:3
>          5: Left Tilt                 , default: Mouse Scroll Left Button    => Left Tilt                 
>              mse, reprogrammable, divertable, persistently divertable, pos:0, group:2, gmask:3
>          6: Right Tilt                , default: Mouse Scroll Right Button   => Right Tilt                
>              mse, reprogrammable, divertable, persistently divertable, pos:0, group:2, gmask:3
>          7: Multiplatform gesture button, default: Multiplatform Gesture Button => Multiplatform gesture button
>              mse, reprogrammable, divertable, persistently divertable, pos:0, group:2, gmask:3
>          8: Virtual Gesture Button    , default: Virtual Gesture Button      => Virtual Gesture Button    
>              divertable, virtual, pos:0, group:3, gmask:0
>      Battery: 50%, discharging, next level 20%.
```

### PR DESCRIPTION
**Pull request by pfps**
Fixes #832

### CODE REVIEW

File: docs/capabilities.md [hunk 1]
```diff
@@ -91,14 +93,17 @@ Settings can only be changed in the Solaar GUI when they are unlocked.
 To unlock a setting click on the icon at the right-hand edge of the setting
 until an unlocked lock appears (with tooltop "Changes allowed").
 
-Solaar keep tracks of the changeable settings of a device.
-Most devices forget changed settings when the are turned off
-or go into a power-saving mode.  When Solaar starts, it restores on-line
-devices to their previously-known state, and while running it restores
+Solaar keep track of most of the changeable settings of a device.
+Devices forget most changed settings when the are turned off
```
Comment (Klap-in) [inline docs/capabilities.md:97]: the=they?

Comment (pfps) [reply docs/capabilities.md:97]: > the=they?

Yes, thanks.

File: docs/features.md [hunk 2]
```diff
@@ -36,20 +36,20 @@ Feature                                | ID       | Status             | Notes
...
+`CHANGE_HOST`                          | `0x1814` | Supported          | `_feature_change_host`
 `HOSTS_INFO`                           | `0x1815` | Partial Support    | `get_host_names`, partial listing only
 `BACKLIGHT`                            | `0x1981` | Unsupported        |
 `BACKLIGHT2`                           | `0x1982` | Supported          | `_feature_backlight2`
 `BACKLIGHT3`                           | `0x1983` | Unsupported        |
 `PRESENTER_CONTROL`                    | `0x1A00` | Unsupported        |
 `SENSOR_3D`                            | `0x1A01` | Unsupported        |
-`REPROG_CONTROLS`                      | `0x1B00` | Listing Only       | `get_keys`, only listing
-`REPROG_CONTROLS_V2`                   | `0x1B01` | Unsupported        |
+`REPROG_CONTROLS`                      | `0x1B00` | Unsupported        | `get_keys`, only listing
```
Comment (Klap-in) [inline docs/features.md:46]: It is right that ‘get_keys’ note is behind the ‘unsupported’?

Comment (pfps) [reply docs/features.md]: No, that should be moved to V2

**Pull request merged**